src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResource.java

```diff
@@ -153,10 +154,27 @@ public Response describeManageDataAccessRequests(@Auth AuthUser authUser) {
     @GET
     @Produces("application/json")
     @Path("/manage/v2")
-    @RolesAllowed({ADMIN, CHAIRPERSON, MEMBER})
-    public Response describeManageDataAccessRequestsV2(@Auth AuthUser authUser) {
-        List<DataAccessRequestManage> dars = dataAccessRequestService.describeDataAccessRequestManageV2(authUser);
-        return Response.ok().entity(dars).build();
+    @RolesAllowed({ADMIN, CHAIRPERSON, MEMBER, SIGNINGOFFICIAL})
+    public Response describeManageDataAccessRequestsV2(@Auth AuthUser authUser, @QueryParam("roleName") Optional<String> roleName) {
+        try {
+            User user = userService.findUserByEmail(authUser.getName());
+            String roleNameValue = roleName.orElse(null);
+            if (Objects.nonNull(roleNameValue)) {
+                //if the roleName is SO and the user does not have that role throw an exception
```

Comment:
Might be helpful to add validation here on the provided role name here so we can throw a different error if the user tries to manage dars for an invalid role name.

Reply:
Okay I just added a validation check for any of the existing user role names that gives a bad request exception - "Signing Official Role is the only role supported at this time" in addition to the other validation check that gives a bad request exception - "Invalid user role name" I also added the nullable enum to the api-docs